### DISCUSSION TITLE
Wiring up external drivers without ramps board

### DISCUSSION THREAD

**Discussion by john-johnston** [fra589/grbl-Mega-5X#200] [Q&A] [answered]
I would like to use larger stepper drivers and bypass the ramps board all together. 
   I am trying to understand how Ramps hands X and Y axis. The step and direction are on analog inputs. Does this mean I should wire the driver with static+5v to the + side of the PUL and DIR driver and then bring the - side of the driver for step and direction to the analog pins?

**Comment by fra589** [comment, marked as answer]
Hi @john-johnston,  
  
The grbl-Megs-5X's default pinout is described in the wiki here: https://github.com/fra589/grbl-Mega-5X/wiki/grbl-Mega-5X-pinout
For connecting externals steppers drivers without using the RAMPS board, you need to connect the 'step', 'direction' (and disable if needed) pins directly to the Arduino pin described in the wiki, 
<kbd>
  <img src="https://user-images.githubusercontent.com/19529154/121039966-d31d8c80-c7b1-11eb-94f9-069ce137d0c0.png">
</kbd>
Example for the X axis, the pins are respectively Arduino's A0, A1 and D38.
  
Finally connect the driver's +5v and GND input to the Arduino's 5v & GND for powering the logic of the stepper..  
  
@++;  
Gauthier.

**Reply by JeffHank1** [reply]
I have also noticed that the Mega board needs to have an external power supply - 12v - if you are driving more than one external stepper motor with the same Step/Dir signal. The USB power alone is not enough. For example, on my CNC I run two external stepper drivers each for the Y axis and A axis. If the Mega board is not externally powered in addition to the USB power supply, the signal gets too weak at times.
Thanks Gauthier for a great program. I am running Grblgru with 5 axes easily.
